### PR TITLE
captured span.sync attribute for ES restclient spans plugins

### LINKED ISSUE #3168: Capture 'sync' attribute on async spans (part2)
This is a follow-up of #3164 

Capture the `span.sync` attribute for the following instrumentation plugins that weren't covered in #3164 :+1: 

- [ ] Elasticsearch
- [ ] Finaggle http client
- [ ] gRPC
- [ ] mongodb (3.x and 4.x)
- [ ] process (commons-exec provides an async execution).

### PR DESCRIPTION
**Pull request by videnkz**
partially covers #3168 

## What does this PR do?

Set the the span.sync = false for asynchronous spans of ES restclient plugins
## Checklist

- [x] This is an enhancement of existing features, or a new feature in existing plugins
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that prove my fix is effective or that my feature works
  - [x] ~~Added an API method or config option? Document in which version this will be introduced~~
  - [x] ~~I have made corresponding changes to the documentation~~

### CODE REVIEW

File: apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static class ElasticsearchRestClientSyncAdvice {
         @Nullable
         @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
         public static Object onBeforeExecute(@Advice.Argument(0) Request request) {
-            return helper.createClientSpan(request.getMethod(), request.getEndpoint(), request.getEntity());
+            return helper.createClientSpan(request.getMethod(), request.getEndpoint(), request.getEntity(), false);
```
Comment (jackshirazi) [inline apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java:63]: shouldn't this be `true`?

Comment (videnkz) [reply apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java]: Sorry, I fixed it.

Comment (jackshirazi) [reply apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java]: shouldn't there have been a test failing from the wrong vaue? Means we're missing tests

Comment (videnkz) [reply apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java]: I added condition that checking `sync attribute` with sync requests.

Comment (videnkz) [reply apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-6_4/src/main/java/co/elastic/apm/agent/esrestclient/v6_4/ElasticsearchClientSyncInstrumentation.java]: Now, if we set wrong value on instrumentation describen above we will get failure testss.

**Review by SylvainJuge — APPROVED** [review]

**Comment by SylvainJuge** [comment]
@Mergifyio update

**Comment by SylvainJuge** [comment]
@Mergifyio queue default

**Comment by SylvainJuge** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by SylvainJuge — APPROVED** [review]

**Comment by SylvainJuge** [comment]
https://github.com/Mergifyio queue default

**Review by JonasKunz — APPROVED** [review]

**Comment by SylvainJuge** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**